airbyte-webapp/src/views/Connection/ConnectionForm/formConfig.tsx

```diff
@@ -254,24 +254,35 @@ const useInitialValues = (
   }, [initialSchema, connection, isEditMode, destDefinition]);
 };
 
-const useFrequencyDropdownData = (): DropDownRow.IDataItem[] => {
+const useFrequencyDropdownData = (
+  additionalFrequency: WebBackendConnectionRead["schedule"]
+): DropDownRow.IDataItem[] => {
   const { formatMessage } = useIntl();
 
-  return useMemo(
-    () =>
-      FrequencyConfig.map((item) => ({
-        value: item.config,
-        label: item.config
-          ? formatMessage(
-              {
-                id: `form.every.${item.config.timeUnit}`,
-              },
-              { value: item.config.units }
-            )
-          : formatMessage({ id: "frequency.manual" }),
-      })),
-    [formatMessage]
-  );
+  return useMemo(() => {
+    const frequencies = FrequencyConfig.map((frequencyConfig) => frequencyConfig.config);
+    if (additionalFrequency) {
+      const additionalFreqAlreadyPresent = frequencies.some(
+        (frequency) =>
+          frequency?.timeUnit === additionalFrequency.timeUnit && frequency?.units === additionalFrequency.units
+      );
+      if (!additionalFreqAlreadyPresent) {
+        frequencies.unshift(additionalFrequency);
+      }
+    }
```

Comment:
This might be a good use-case for a [set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set)
But we'd need to sort on return.

Reply:
I think given that we know our frequency config is rather short (I mean we'll never talk about more than 100 entries here), converting this into a set (and handling the sorting ourselves) sounds like way more effort (even in terms of performance), than just iterating through the (rather small array) once.